### DISCUSSION TITLE
how to use Poco::Timespan correctly ?

### DISCUSSION THREAD

**Discussion by omerbrandis** [pocoproject/poco#4080] [Q&A]
Hello all,

I'm somehow unable to use Poco::Timespan.

```
FullPickupDateTime  = Poco::DateTime(PickupDate.year(),PickupDate.month(),PickupDate.day(),PickupTime.hour(),PickupTime.minute());
FullDropOffDateTime = Poco::DateTime(DropOffDate.year(),DropOffDate.month(),DropOffDate.day(),DropOffTime.hour(),DropOffTime.minute());
Poco::Timespan TS   = FullDropOffDateTime   - FullPickupDateTime ;
std::cout << FullDropOffDateTime.day() << "-" << FullPickupDateTime.day() <<  " hours diff " << TS.hours()<< std::endl;

```    
writes to stdout:
12-11 hours diff 0
which is the correct/expected value for FullDropOffDateTime.day() and FullPickupDateTime.day(), 
but not for TS.hours() which should be 24.

while using Poco::Timestamp::TimeDiff on the same variables, works as expected :
```
Poco::Timestamp::TimeDiff TD = FullDropOffDateTime.timestamp()   - FullPickupDateTime.timestamp() ;
std::cout << "hours diff " << TD/1000000/60/60 << std::endl;
```
writes to stdout :
hours diff 24
which is the expected result.

running poco 11.1.

please advise,
Omer.

**Comment by micheleselea** [comment]
I think you have to user [totalHours](https://docs.pocoproject.org/current/Poco.Timespan.html#20403)
